engine/classes/Elgg/Database/EntityTable.php

```diff
@@ -217,13 +296,29 @@ public function get($guid, $type = '') {
 			return false;
 		}
 
-		// Check local cache first
-		$new_entity = _elgg_services()->entityCache->get($guid);
-		if ($new_entity) {
+		$guid = (int) $guid;
+
+		$memcache = _elgg_get_memcache('new_entity_cache');
+		
+		$entity = $this->entity_cache->get($guid);
+		if (!$entity) {
+			// Check caches
+			$entity = $memcache->load($guid);
+			if ($entity instanceof ElggEntity) {
+				// until ACLs in memcache, DB query is required to determine access
+				if (!$this->getRow($guid)) {
```

Comment:
Let's remove this and check for ignored access || has_access_to_entity below. We need to make sure entity is accesible even if it comes from in-memory cache.